addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -355,33 +393,71 @@ public void notifyCallback(String challenge, HttpMessage targetMessage) {
+                    try {
+                        HttpMessage oastReceivedMsg =
+                                request.getHistoryReference().getHttpMessage();
+                        if (oastReceivedMsg
+                                .getRequestHeader()
+                                .getURI()
+                                .toString()
+                                .contains(payloadUriHost)) {
+                            for (HttpMessage msg : attackMsgs) {
+                                if (msg.getHistoryRef() != null) {
+                                    Optional<Alert> optionalAlert =
+                                            msg.getHistoryRef().getAlerts().stream()
+                                                    .filter(
+                                                            alert ->
+                                                                    (PLUGIN_ID
+                                                                                    + QUALIFIER_OUT_OF_BAND)
+                                                                            .equals(
+                                                                                    alert
+                                                                                            .getAlertRef()))
+                                                    .findFirst();
+                                    if (optionalAlert.isPresent()) {
+                                        Alert existingAlert = optionalAlert.get();
+                                        String info =
+                                                existingAlert.getOtherInfo()
+                                                        + "Request\n"
+                                                        + msg.getRequestHeader()
+                                                        + msg.getRequestBody()
+                                                        + "\nResponse\n"
+                                                        + msg.getResponseHeader()
+                                                        + msg.getResponseBody()
+                                                        + "\n--------------------------------\n";
+                                        existingAlert.setOtherInfo(info);
+                                        msg.getHistoryRef().updateAlert(existingAlert);
+                                        continue;
+                                    }
+                                }
+                                newAlert()
+                                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                                        .setAttack(msg.getRequestBody().toString())
+                                        .setEvidence(oastReceivedMsg.getRequestHeader().toString())
+                                        .setMessage(msg)
+                                        .setAlertRef(PLUGIN_ID + QUALIFIER_OUT_OF_BAND)
```

Comment:
This is a neat idea, to help us know which "check" triggered the alert. Nice move!

Reply:
The purpose of alert references was to differentiate alert types, not same alert with different checks. This will have implications in the documentation, which creates an entry per alert reference.